### PR TITLE
Enable mergify

### PR DESCRIPTION
**Pull request by kinnou02**
Automatically merge PR done by @CanalTP/coreteam that have
  - at least 2 approved reviews
  - no comment or reuqest for change reviews
  - all requested reviewver have approved
  - all CI tests are ok
  - no label wip, dot_not_merge, etc...

PR are rebased on `dev` if needed and tests are rerun before merge.

Reviews are dismissed when a push occurs.

### CODE REVIEW

**Review by AurelienLP — APPROVED** [review]

**Review by pbougue — COMMENTED** [review]
This conf is the best I can think of (a bit of delay might be good but not doable).
I still vote to wait for a test on other repos, but I will follow the move if the majority is onboard!

**Review by pbougue — COMMENTED** [review]
Can we do something about sub-repos needing a merge (then update parent repo) before actual merge?
Just add do-not-merge is OK, if we can think of some :robot:-thing for that it would be better.

**Comment by kinnou02** [comment]
> Can we do something about sub-repos needing a merge (then update parent repo) before actual merge?
> Just add do-not-merge is OK, if we can think of some -thing for that it would be better.

This should be handled by ci, and I think @nonifier did something about this

**Review by benoit-bst — APPROVED** [review]

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - status-success=continuous-integration/travis-ci/pr
+      - status-success=default
+      - status-success=codecov/project
+      - status-success=codecov/patch
+      - base=dev
+      - label!=dot_not_merge
+      - label!=wip
+      - label!=waiting
+      - label!=merge_after_release
+      - author=@CoreTeam
```
Comment (datanel) [inline .mergify.yml]:
```suggestion
      - author=@coreteam
```

Comment (datanel) [reply .mergify.yml]: this  should be the id, not the name (tested in https://github.com/CanalTP/transit_model)

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+pull_request_rules:
+  - name: automatic merge when CI passes and 2 reviews
+    conditions:
+      - "#approved-reviews-by>=2"
+      - "#review-requested=0"
+      - "#changes-requested-reviews-by=0"
+      - "#commented-reviews-by=0"
```
Comment (datanel) [inline .mergify.yml:7]: if you approve and then comment, you (all the reviewers) must re-approve

**Review by nonifier — APPROVED** [review]

**Pull request merged**